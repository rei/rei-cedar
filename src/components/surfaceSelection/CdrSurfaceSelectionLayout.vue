<script setup lang="ts">
import { useCssModule, computed } from 'vue';
import mapClasses from '../../utils/mapClasses';
import { surfaceSelectionLayout } from '../../types/interfaces';

/** Component for layout of a surface selection */

defineOptions({ name: 'CdrSurfaceSelectionLayout' });

const props = withDefaults(defineProps<surfaceSelectionLayout>(), {
  orientation: 'horizontal',
});

const style = useCssModule();
const baseClass = 'cdr-surface-selection-layout';

// Manages the props passed along to root
const rootProps = computed(() => {
  const { orientation } = props;
  const classes = [baseClass];

  // Add orientation
  classes.push(`${baseClass}--${orientation}`);

  return { class: mapClasses(style, ...classes) };
});
</script>

<template>
  <div v-bind="rootProps">
    <slot />
  </div>
</template>

<style lang="scss" module src="./styles/CdrSurfaceSelectionLayout.module.scss"></style>
