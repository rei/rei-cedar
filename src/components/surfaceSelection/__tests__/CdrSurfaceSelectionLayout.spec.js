import { mount } from '../../../../test/vue-jest-style-workaround.js';
import CdrSurfaceSelectionLayout from '../CdrSurfaceSelectionLayout.vue';

describe('CdrSurfaceSelectionLayout', () => {
  describe('snapshot test', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(CdrSurfaceSelectionLayout, { props: { orientation: 'horizontal' } });
    });

    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('orientation test', () => {
    it('shows horizontal', () => {
      const wrapper = mount(CdrSurfaceSelectionLayout, { props: { orientation: 'horizontal' } });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.classes()).toEqual([
        'cdr-surface-selection-layout',
        'cdr-surface-selection-layout--horizontal',
      ]);
    });

    it('shows vertical', () => {
      const wrapper = mount(CdrSurfaceSelectionLayout, { props: { orientation: 'vertical' } });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.classes()).toEqual([
        'cdr-surface-selection-layout',
        'cdr-surface-selection-layout--vertical',
      ]);
    });
  });
});
