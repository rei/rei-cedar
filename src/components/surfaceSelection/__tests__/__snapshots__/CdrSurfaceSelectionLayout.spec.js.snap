// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CdrSurfaceSelectionLayout > orientation test > shows horizontal 1`] = `
<div
  class="cdr-surface-selection-layout cdr-surface-selection-layout--horizontal"
>
  
  
</div>
`;

exports[`CdrSurfaceSelectionLayout > orientation test > shows vertical 1`] = `
<div
  class="cdr-surface-selection-layout cdr-surface-selection-layout--vertical"
>
  
  
</div>
`;

exports[`CdrSurfaceSelectionLayout > snapshot test > renders correctly 1`] = `
<div
  class="cdr-surface-selection-layout cdr-surface-selection-layout--horizontal"
>
  
  
</div>
`;
